tls/s2n_fingerprint.h

```diff
@@ -22,10 +22,25 @@
 #include "tls/s2n_client_hello.h"
 #include "utils/s2n_result.h"
 
+struct s2n_fingerprint {
+    size_t raw_size;
+    const struct s2n_fingerprint_method *method;
+    struct s2n_client_hello *client_hello;
+    struct s2n_hash_state hash;
+    /* Originally we represented the hash as the raw bytes of the digest.
+     * However, the JA4 hash is composed of a string prefix and two digests,
+     * so cannot reasonably be represented as raw bytes.
+     * Keep the old behavior for the old method.
+     */
+    unsigned int legacy_hash_format : 1;
```

Comment:
I don't love this. The other options I considered:
* Remove the old method. After all, it is marked unstable.
* Translate the hex back to raw bytes in the old method. That seemed messy, since we'd translate to hex just to immediately translate back, but maybe that's better than these flags.

Reply:
Hmm, personally I like the second option? I prefer complicating the legacy method to complicating the new method.

Reply:
I think I agree, but I'd prefer to fix it in a follow-up PR.

Customers are actively using the legacy stuff, but not actively using the new stuff. snprintf / sscanf are a quick way to handle hex without complicating the code, but they're not exactly efficient. With a quick timing test, I've got the same number of operations taking 12s with the extra unnecessary snprintf + sscanf that took 9s before. Not a crazy slowdown, but also really great.

I can get it down to 10s if I use the stuffer hex methods (+ LTO), but the stuffer hex methods are currently only for testing. Making them available for non-testing would complicate this PR. I can also actually keep the timing at 9s if I inline my own hex conversion logic, so I'd also want to investigate whether the stuffer hex methods are doing something inefficient.